### PR TITLE
Feature/dropdown menu

### LINKED ISSUE #534: [S38.1]: Menu dropdown/up
### Link to Epic

Epic: https://github.com/Open-Study-College/osc/issues/171

### Description

n/a

### Linked Tasks

#535 

_No response_

### LINKED ISSUE #535: Build dropdown component
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/534

### Description

n/a

### PR DESCRIPTION
**Pull request by mattwendzina**
Closes #534 
Closes #535 

## 📝 Description

Add Dropdown Menu component

## ⛳️ Current behavior (updates)

N/A

## 🚀 New behavior

Dropdown menu component added

### CODE REVIEW

**Review by stevenbister — CHANGES_REQUESTED** [review]
Would it be possible to change the functionality of the checkbox and radio options slightly? 

I don't really like how it closes the menu when you select something. I would expect to see the UI update so I know what I have selected and for checkboxes I might want to select multiple options -- having to click back into the menu each time is kinda annoying.

It looks like there's an onSelect prop for the DropdownMenuItem that if you pass `e.preventDefault()` to will prevent this behaviour
https://www.radix-ui.com/docs/primitives/components/dropdown-menu#item

File: packages/osc-ui/src/components/DropdownMenu/DropdownMenu.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/**
+ * @vitest-environment jsdom
+ */
```
[the remark this answers is not in the document]
Comment (mattwendzina) [reply packages/osc-ui/src/components/DropdownMenu/DropdownMenu.spec.tsx]: Done 👍

**Comment by mattwendzina** [comment]
> Would it be possible to change the functionality of the checkbox and radio options slightly?
> 
> I don't really like how it closes the menu when you select something. I would expect to see the UI update so I know what I have selected and for checkboxes I might want to select multiple options -- having to click back into the menu each time is kinda annoying.
> 
> It looks like there's an onSelect prop for the DropdownMenuItem that if you pass `e.preventDefault()` to will prevent this behaviour https://www.radix-ui.com/docs/primitives/components/dropdown-menu#item

Good idea. Added this in and seems to work fine 👍

**Review by stevenbister — APPROVED** [review]
I'm happy with that 😄

**Review by hi-im-si — APPROVED** [review]

**Pull request merged**